crates/core/src/logstore/storage/cache.rs

```diff
@@ -0,0 +1,327 @@
+            _ => return Ok(None),
+        }
+        Ok(Some(()))
+    }
+}
+
+async fn get_default_cache(
+    config: LogCacheConfig,
+) -> DeltaResult<(DeltaLogCache, Option<TempDir>), CachedStoreError> {
+    let dir = tempfile::tempdir()?;
+
+    Ok((
+        Arc::new(
+            HybridCacheBuilder::new()
+                .memory(config.max_size_memory_mb * 1024 * 1024)
+                .storage(Engine::Large)
+                .with_device_options(
+                    DirectFsDeviceOptions::new(dir.path())
+                        .with_capacity(config.max_size_disk_mb * 1024 * 1024),
+                )
+                .build()
+                .await
+                .map_err(|_| CachedStoreError::CacheInitialization)?,
+        ),
+        Some(dir),
+    ))
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+struct Entry {
+    data: Bytes,
+    last_modified: DateTime<Utc>,
+    e_tag: Option<String>,
+}
+
+impl Entry {
+    fn new(data: Bytes, last_modified: DateTime<Utc>, e_tag: Option<String>) -> Self {
+        Self {
+            data,
+            last_modified,
+            e_tag,
+        }
+    }
+}
+
+/// An object store implementation that conditionally caches file requests.
+///
+/// This implementation caches the file requests based on on the evaluation
+/// of a condition. The condition is evaluated on the path of the file and
+/// can be configured to meet the requirements of the user.
+///
+/// This is __not__ a general purpose cache and is specifically designed to cache
+/// the commit files of a Delta table. E.g. it is assumed that files written to
+/// the object store are immutable and no attempt is made to invalidate the cache
+/// when files are updated in the remote object store.
+pub(super) struct CommitCacheObjectStore {
+    /// The wrapped object store.
+    inner: Arc<dyn ObjectStore>,
+
+    /// Hybrid cache that stores entries in memory and on disk.
+    cache: DeltaLogCache,
+
+    /// Fully qualified URL of the object store.
+    ///
+    /// This is required to avoid collisions with other cache files
+    /// when the cache is used across multiple instances.
+    root_url: url::Url,
+
+    /// Directory where the cache files are stored.
+    ///
+    /// We need to keep a reference to the temporary directory to ensure that it is
+    /// not deleted while the cache is still in use - i.e. TempDir will be deleted
+    /// when the CommitCacheObjectStore is dropped.
+    cache_dir: Option<TempDir>,
+}
+
+impl std::fmt::Debug for CommitCacheObjectStore {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("CommitCacheObjectStore")
+            .field("object_store", &self.inner)
+            .finish()
+    }
+}
+
+impl std::fmt::Display for CommitCacheObjectStore {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "CommitCacheObjectStore({})", self.inner)
+    }
+}
+
+fn cache_delta(path: &Url) -> bool {
```

Comment:
Small nit to improve readability
```suggestion
fn is_delta_commit(path: &Url) -> bool {
```